streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -203,7 +203,7 @@ public void process(final Record<KIn, VIn> record) {
         } catch (final FailedProcessingException | TaskCorruptedException | TaskMigratedException e) {
             // Rethrow exceptions that should not be handled here
             throw e;
-        } catch (final Exception e) {
+        } catch (final RuntimeException e) {
```

Comment:
Side cleanup -- no reason to catch `Exception` as no checked exceptions are declared (same below)